### DISCUSSION TITLE
Meets a lot of error "MissingGreenlet greenlet_spawn has not been called; can't call await_only() here"

### DISCUSSION THREAD

**Discussion by YadongM** [sqlalchemy/sqlalchemy#12007] [Usage Questions]
Hi sqlalchemy, I found a lot of this kind Error in my project. 
There's no application code in this stack trace at all. 

```
exception.traceback	
File "/usr/local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 763, in _finalize_fairy
    fairy._reset(pool, transaction_was_reset)
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 1038, in _reset
    pool._dialect.do_rollback(self)
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 683, in do_rollback
    dbapi_connection.rollback()
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 753, in rollback
    self._handle_exception(error)
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 684, in _handle_exception
    raise error
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 747, in rollback
    self.await_(self._rollback_and_discard())
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 61, in await_only
    raise exc.MissingGreenlet(

exception.type	
MissingGreenlet

exception.value	
greenlet_spawn has not been called; can't call await_only() here. Was IO attempted in an unexpected place? (Background on this error at: https://sqlalche.me/e/14/xd2s)
```

Maybe this is because of the connection pool? Here is my connection pool code lines:

```
async_engine = create_async_engine(
    configs.async_sqlalchemy_url,
    pool_size=25,
    max_overflow=10,
    pool_timeout=30,
    pool_pre_ping=True,
)
Async_session = sessionmaker(
    autocommit=False,
    autoflush=False,
    bind=async_engine,
    class_=AsyncSession,
    expire_on_commit=False,
)
```

And I changed the db parameter:

>idle_in_transaction_session_timeout= 120000
idle_session_timeout=300000

Here is dependence version information.

>asyncpg == 0.29.0
fastapi == 0.110.0
fastapi-utils == 0.2.1
SQLAlchemy == 1.4.54

**Comment by CaselIT** [comment]
Hi,

That seems to be caused by a connection (or session) going out of scope without being closed.
That's not supported when using asyncio drivers, hence the error.

Ensure that you either close the connections and session manually (calling `await thing.close()`) or use them in `async with` statements so that they are properly closed

**Comment by tommy-bump-be** [comment]
I meet this issue when I try to session close without release reference. 
If you session close earlier than object which is refer to same session, it generate missing greenlet exception. 
you can handle this issue with weak reference or `del` code(delete in memory).

**Reply by CaselIT** [reply]
I don't think that's correct. You get that only if the object tries to lazy load something, that's not possible when using asyncio
